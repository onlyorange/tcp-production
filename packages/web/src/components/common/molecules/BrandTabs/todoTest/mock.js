export default {
  brand_tabs: [
    {
      id: '1',
      url: '/',
      alt: "The Children's Place",
      title: "The Children's Place",
      target: '',
      logoClass: 'tcp_logo_medium',
      active: true,
    },
    {
      id: '2',
      url: 'https://www.gymboree.com/',
      alt: 'Gymboree',
      title: 'Gymboree',
      target: '_blank',
      logoClass: 'gymboree_logo_medium',
      active: false,
    },
  ],
};
