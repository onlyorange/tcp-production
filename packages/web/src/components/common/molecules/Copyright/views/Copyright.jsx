import React from 'react';
import PropTypes from 'prop-types';
import RichText from '@tcp/core/src/components/common/atoms/RichText/views/RichText';
import withStyles from '@tcp/core/src/components/common/hoc/withStyles';
import { getLocator } from '@tcp/web/src/utils';
import style from '../Copyright.style';

const Copyright = ({ className, children }) => (
  <React.Fragment>
    <RichText
      dataLocator={getLocator('content-copyright')}
      className={className}
      richTextHtml={children}
    />
  </React.Fragment>
);

Copyright.propTypes = {
  className: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

export { Copyright as CopyrightVanilla };
export default withStyles(Copyright, style);
