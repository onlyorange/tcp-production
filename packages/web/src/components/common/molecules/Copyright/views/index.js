export { default, CopyrightVanilla } from './Copyright';
