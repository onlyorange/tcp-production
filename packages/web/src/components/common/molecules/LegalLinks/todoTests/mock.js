export default {
  links: [
    {
      url: 'https://www.childrensplace.com/us/help-center/#termsAndConditionsli',
      title: 'Terms and Conditions title',
      text: 'Terms and Conditions',
      target: '',
    },
    {
      url: 'https://www.childrensplace.com/us/help-center/#privacyPolicySectionli',
      title: 'Privacy Policy title',
      text: 'Privacy Policy',
      target: '',
    },
    {
      url: 'https://www.childrensplace.com/us/help-center/#cookiePolicySectionli',
      title: 'Cookie Policy title',
      text: 'Cookie Policy',
      target: '',
    },
    {
      url: 'https://www.childrensplace.com/us/content/supply-chain',
      title: 'California supply chain act title',
      text: 'California Supply Chains Act',
      target: '',
    },
    {
      url: 'https://www.childrensplace.com/us/sitemap',
      title: 'Site Map title',
      text: 'Site Map',
      target: '',
    },
  ],
};
