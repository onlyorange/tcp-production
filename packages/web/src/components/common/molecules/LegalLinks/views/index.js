export { default, LegalLinksVanilla } from './LegalLinks';
