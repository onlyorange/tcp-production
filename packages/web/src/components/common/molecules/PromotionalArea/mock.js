export default {
  promo_message_wrapper: [
    {
      id: '1',
      url: '/static/promo',
      text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 1.',
      title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 1.',
      target: '_blank',
      class: 'promo_message',
    },
    {
      id: '2',
      url: '/static/promo2',
      text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 2.',
      title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 2.',
      target: '_blank',
      class: 'promo_message',
    },
    {
      id: '3',
      url: '/static/promo3',
      text: '<b>FREE SHIPPING EVERY DAY!</b> No minimum purchase required 3.',
      title: 'FREE SHIPPING EVERY DAY! No minimum purchase required 3.',
      target: '_blank',
      class: 'promo_message',
    },
  ],
};
