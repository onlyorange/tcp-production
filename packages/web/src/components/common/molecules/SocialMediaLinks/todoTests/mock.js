export default {
  connectWithUsLabel: '',
  links: [
    {
      url: 'http://facebook.com/childrensplace',
      title: 'Facebook',
      target: '_blank',
      iconClass: 'icon-facebook',
    },
    {
      url: 'http://twitter.com/childrensplace',
      title: 'Twitter',
      target: '_blank',
      iconClass: 'icon-twitter',
    },
    {
      url: 'http://instagram.com/childrensplace',
      title: 'Instagram',
      target: '_blank',
      iconClass: 'icon-instagram',
    },
    {
      url: 'http://pinterest.com/childrensplace',
      title: 'Pinterest',
      target: '_blank',
      iconClass: 'icon-pinterest',
    },
  ],
};
