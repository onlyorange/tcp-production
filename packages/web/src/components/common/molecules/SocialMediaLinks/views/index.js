export { default, SocialMediaLinksVanilla } from './SocialMediaLinks';
