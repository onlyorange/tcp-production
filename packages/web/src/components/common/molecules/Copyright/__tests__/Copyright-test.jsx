import React from 'react';
import { shallow } from 'enzyme';
import { CopyrightVanilla } from '../views';
import mock from './mock';

describe('Copyright component', () => {
  it('renders correctly', () => {
    const props = {
      className: 'test-copyright',
    };
    const component = shallow(<CopyrightVanilla {...props}>{mock.text}</CopyrightVanilla>);
    expect(component).toMatchSnapshot();
    expect(component.find('.test-copyright').html()).toContain(mock.text);
  });
});
