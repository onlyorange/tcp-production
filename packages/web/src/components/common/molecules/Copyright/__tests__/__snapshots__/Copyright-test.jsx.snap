// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Copyright component renders correctly 1`] = `ShallowWrapper {}`;
